File: PetJava/src/main/java/com/pet/entity/Notification.java
package com.pet.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("notification")
public class Notification {
    @TableId(type = IdType.ASSIGN_UUID)
    private String id;
    
    private String userId;
    private String type;  // booking/payment/health/review/system/message
    private String title;
    private String content;
    private Boolean isRead;
    private String link;
    private String data;  // JSON
    
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    @TableLogic
    private Integer deleted;
}

File: PetVue/src/types/notification.ts
export interface Notification {
  id: string
  userId: string
  type: NotificationType
  title: string
  content: string
  relatedId?: string
  isRead: boolean
  createdAt: Date
}

export type NotificationType =
  | 'booking_created'
  | 'booking_confirmed'
  | 'booking_cancelled'
  | 'health_update'
  | 'health_alert'
  | 'review_reminder'
  | 'message'
  | 'system'

export interface NotificationPreferences {
  email: NotificationChannelPrefs
  push: NotificationChannelPrefs
  inApp: NotificationChannelPrefs
}

export interface NotificationChannelPrefs {
  booking: boolean
  health: boolean
  message: boolean
  system: boolean
}
